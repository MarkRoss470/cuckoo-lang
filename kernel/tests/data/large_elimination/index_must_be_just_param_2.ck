data Nat : Type where
  | zero : Nat
  | succ : Nat -> Nat

data SomeProp : Nat -> Prop where
  | c : (n : Nat) -> SomeProp (Nat.succ n)

-- Even though `succ` is an injection, `SomeProp` is not large eliminating because `n` is not used verbatim in an index
def some_prop_rec : Prop := SomeProp.rec.{1}