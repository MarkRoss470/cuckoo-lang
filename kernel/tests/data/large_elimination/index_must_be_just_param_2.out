ERROR at 9:29-9:45: SomeProp.rec takes 0 level argument(s), but 1 were provided.